CovertInfiltration/Src/CovertInfiltration/Classes/XComGameState_ActivityChainSpawner.uc

```diff
@@ -0,0 +1,451 @@
+//---------------------------------------------------------------------------------------
+//  AUTHOR:  Xymanek
+//  PURPOSE: Replacement for base game MissionCalendar which instead spawns CAs based on
+///          "work" done by XCom (base) and contacted regions and relays built (bonus)
+//---------------------------------------------------------------------------------------
+//  WOTCStrategyOverhaul Team
+//---------------------------------------------------------------------------------------
+
+class XComGameState_ActivityChainSpawner extends XComGameState_BaseObject config(Infiltration);
+
+var protectedwrite float PreviousWork;
+var protectedwrite TDateTime PreviousWorkSubmittedAt;
+
+// Work rate is meaured in hours
+var protectedwrite int CachedWorkRate;
+var protectedwrite int NextSpawnAt; // In work units
+
+var array<ChainDeckEntry> ChainDeck;
+var bool bDarkEventsActive;
```

Comment:
This will be persisted into history, and `ChainDeckEntry` has an `X2ActivityChainTemplate` member - object references should not be saved to history